gm4_damaging_stonecutters/data/damaging_stonecutters/functions/damage.mcfunction

```diff
@@ -0,0 +1,14 @@
+#damages entity
+#@s - @e[type=!#damaging_stonecutters:stonecutter_immune]
+#called by damaging_stonecutters:check_damage
+
+#damage entity
+effect give @s minecraft:absorption 1 0 true
+effect give @s[type=!#damaging_stonecutters:undead] minecraft:instant_damage 1 0 true
+effect give @s[type=#damaging_stonecutters:undead] minecraft:instant_health 1 0 true
+
+#audiovisuals
+particle minecraft:block minecraft:stonecutter ~ ~ ~ .25 0 .25 0 10 normal @a
+particle minecraft:block minecraft:redstone_block ~ ~ ~ .25 0 .25 0 10 normal @a
```

Comment:
I'm not to sure about the "blood" particles used here. No other source of damage in the game uses such an effect.

Reply:
I agree with you here, someone else suggested it (I can't remember who). Should I remove the stonecutter particle too?